### PR TITLE
Extend `parseCsv` to `http` and `bigquery`

### PR DESCRIPTION
**Pull request by mtuchi**
## Summary

Use  `common` `parseCsv` in `http` and `bigquery`

## Review Checklist

Before merging, the reviewer should check the following items:

- [x] Does the PR do what it claims to do?
- [ ] If this is a new adaptor, has the migration tool been run and the
      migration guide followed?
- [ ] Are there any unit tests? Should there be?
- [x] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.

### CODE REVIEW

**Review by taylordowns2000 — APPROVED** [review]

**Comment by taylordowns2000** [comment]
I've added a broken test in here and want to hold off on the merge until I get the right idea !

@josephjclark , @stuartc ...

Mtuchi and I couldn't _quite_ figure out how to use `parseCsv` in this canonical implementation (see failing test in language-http) where we want to send an http request for each chunk and see the responses from those http requests in state.

If the answer is that we can make the requests (which we can, according to my new test), but that we can't see the responses in state, that might also be ok.

But I want to make sure I understand the limits here before merging. Thank you!

File: packages/http/test/index.js
Comment (josephjclark) [inline packages/http/test/index.js:543]: @taylordowns2000 Coming in out of context so I may be missing something, but isn't the http response now on state.data here?

So you can just `state.apiReferences.push(state.data)`

If we're not seeing that at runtime, that feels like a bug

File: packages/http/test/index.js
Comment (josephjclark) [inline packages/http/test/index.js:536]: If you're looking for a more functional solution, you could declare the callback like this:

```
(state, rows) => rows.length && post(...)
```
You wouldn't need to async/await the post because it'll return a promise.

Comment (mtuchi) [reply packages/http/test/index.js:536]: @josephjclark i tried this approach 
```
(state, rows) =>
  rows.length > 0 &&
  post(
    'https://www.example.com/api/csv-reader',
    {
      body: rows,
    },
    state => {
      console.log("I'm a regular callback");
      return state;
    }
  )
``` 
It did not work. i was getting this error 
```
 AssertionError: expected [Function] to deeply equal { data: { references: [], …(2) } }
 ```
 
 Then tried 
 ```
 (state, rows) =>
  rows.length > 0 &&
  post(
    'https://www.example.com/api/csv-reader',
    {
      body: rows,
    },
    state => {
      console.log("I'm a regular callback");
      return state;
    }
  )(state)
  ```
  Now i am getting this error 
  ```
   TypeError: state.references is not iterable
   ```
   I think we should pair to see pair to discuss this further, You can see the [current implementation of parseCsv here ](https://github.com/OpenFn/adaptors/blob/7034947eded80f6dc69681be3ec61a7dd38dbcd8/packages/common/src/Adaptor.js#L565)

**Comment by mtuchi** [comment]
@josephjclark i have pushed the changes from our pairing today. Can you review this PR and let me know if it needs more adjustment

File: packages/common/src/Adaptor.js
Comment (josephjclark) [inline packages/common/src/Adaptor.js:494]: Is it correct that we're publicly documenting  `splitKeys` now?

I'm not against it but it doesn't feel related to this work. It should really be mentioned in the main PR description.

File: packages/common/src/Adaptor.js
Comment (josephjclark) [inline packages/common/src/Adaptor.js:633]: Hmm, this shouldn't be removed here either.

Is this diff just out of date? If we're behind then that's ok :)

File: packages/http/test/index.js
Comment (josephjclark) [inline packages/http/test/index.js:516]: Should probably remove this log

File: packages/http/test/index.js
Comment (josephjclark) [inline packages/http/test/index.js:527]: @taylordowns2000 just to bring you up to speed on this, here's what your http test looks like now.

Is this all as you expect?

**Review by josephjclark — CHANGES_REQUESTED** [review]
Dependencies need updating in bigquery and http

File: packages/common/src/Adaptor.js
Comment (stuartc) [reply packages/common/src/Adaptor.js:494]: This was me, I guess we were exporting it and _not_ documenting it - so I thought we should document the interface.

@mtuchi can you update the PR notes to include something like "Added documentation for `splitKeys`.

File: packages/http/test/index.js
Comment (taylordowns2000) [reply packages/http/test/index.js:526]: @mtuchi is all good with this! thanks joe!

**Review by josephjclark — APPROVED** [review]

**Review by josephjclark — APPROVED** [review]
Looks good to me now!

**Pull request merged**
